web/packages/teleterm/src/ui/DocumentGatewayApp/AppGateway.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useMemo, useRef } from 'react';
+
+import { Flex, Text, ButtonSecondary, Link, Box, Alert } from 'design';
+
+import Validation from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAsync';
+import { debounce } from 'shared/utils/highbar';
+
+import { copyToClipboard } from 'design/utils/copyToClipboard';
+
+import { PortFieldInput } from 'teleterm/ui/DocumentGateway/common';
+import { Gateway } from 'teleterm/services/tshd/types';
+import { CliCommand } from 'teleterm/ui/DocumentGateway/CliCommand';
+
+export function AppGateway(props: {
+  gateway: Gateway;
+  disconnectAttempt: Attempt<void>;
+  changePort(port: string): void;
+  changePortAttempt: Attempt<void>;
+  onDisconnect(): void;
+}) {
+  const formRef = useRef<HTMLFormElement>();
+  const cliCommandPreview = props.gateway.gatewayCliCommand.preview;
+
+  const { changePort } = props;
+  const handleChangePort = useMemo(() => {
+    return debounce((value: string) => {
+      if (formRef.current.reportValidity()) {
+        changePort(value);
+      }
+    }, 1000);
+  }, [changePort]);
+
+  return (
+    <Box maxWidth="680px" width="100%" mx="auto" mt="4" px="5">
+      <Flex justifyContent="space-between" mb="4" flexWrap="wrap" gap={2}>
+        <Text typography="h3">App Connection</Text>
+        {props.disconnectAttempt.status === 'error' && (
+          <Alert>
+            Could not close the connection: {props.disconnectAttempt.statusText}
+          </Alert>
+        )}
+        <ButtonSecondary size="small" onClick={props.onDisconnect}>
+          Close Connection
+        </ButtonSecondary>
+      </Flex>
+      <Text typography="h4" mb={1}>
+        Local app proxy
+      </Text>
```

Comment:
I think we should remove this header. We don't mention local proxies anywhere else in the UI, so we should probably not introduce new terminology here.

Reply:
Yeah, I was unsure about this header, in the end I decided to include it. I'm fine with removing it 👍